_posts/2019-05-28-experience-and-opinions-developing-for-google-assistant-and-amazon-alexa.md

```diff
@@ -0,0 +1,92 @@
+
+### Beyond Voice
+{: .text-left }
+
+Digital assistants lasted without screens for barely a year. Skills can now play videos, show images, and present content in basic HTML components to compliment the voice responses. Alexa has a presentation language [APL](https://developer.amazon.com/docs/alexa-presentation-language/apl-overview.html) to design for this but had limited SDK support. The Google Actions SDK for nodejs had cards, buttons, tables, images, etc all built in. It was simple and fun to add the visual components while developing. The simulator displays everything nicely.
+
+## Testing
+{: .text-left }
+
+I really like the ask CLI way of testing a skill. Talking to your skill over and over again is bad and typing it over and over is worse. The ask CLI supports simple generated [text file recordings](https://github.com/agrc/digital-assisant-skills/blob/alexa-skill/recordings/all.template) that direct the conversation in the test. The CLI outputs the responses and you can visually approve the response. I could not find a way to do this in the Google Assistant simulator. The node sdk does have a concept call `Suggestions` that allow you to click a suggestion bubble instead of typing and that was an improvement but not as nice as the recordings.
+
+### Location
+{: .text-left }
+
+The Google Assistant and Alexa simulators are very nice. They give you screens to test presentation markup as well as other features. One feature that the Alexa simulator is missing is location. You cannot set a location for the simulator to use. Google makes this very simple and it is on the main UI. A major difference between the two assistants is that addresses are assigned to devices in Amazon. Amazon is thinking, where do I ship you all of your Holiday packages. The Google Assistant is on the every Android phone so location is where am I now.
+
+## Resources
+{: .text-left }
+
+### Documentation
+{: .text-left }
+
+There are so many pages of documentation written for the Google Assistant and Amazon Alexa. It is a massive undertaking. There are multiple versions of the same documentation for each SDK language and then the underlying request response contracts if you decide to roll your own. With that said, I found Alexa's documentation and demos to be better organized and more current. The ecosystem seemed to be treated more as one item. Google seemed to be offering different products that worked together and the documentation did not seem as cohesive. A lot of the demo code was on older versions of their SDK's.
```

Comment:
```suggestion
There are so many pages of documentation written for both Google Assistant and Amazon Alexa. It is a massive undertaking. There are multiple versions of the same documentation for each SDK language and then the underlying request response contracts if you decide to roll your own. With that said, I found Alexa's documentation and demos to be better organized and more current. The Alexa ecosystem seemed to be treated more as one item, while Google seemed to be offering different products that worked together, so the documentation did not seem as cohesive. A lot of the Google demo code was on older versions of their SDKs.
```

Reply:
What is a massive undertaking? To read through the documentation? Can we clarify here?